Assets/Scripts/Animator/ModelAnimator.cs

```diff
@@ -12,19 +12,115 @@
+    _joints[Landmarks.LeftAnkle] = new Joint(_anim.GetBoneTransform(HumanBodyBones.LeftFoot));
+    _joints[Landmarks.LeftFootIndex] = new Joint(_anim.GetBoneTransform(HumanBodyBones.LeftToes));
+
+    // Misc
+    _joints[Landmarks.Hip] = new Joint(_anim.GetBoneTransform(HumanBodyBones.Hips));
+    _joints[Landmarks.Head] = new Joint(_anim.GetBoneTransform(HumanBodyBones.Head));
+    _joints[Landmarks.Neck] = new Joint(_anim.GetBoneTransform(HumanBodyBones.Neck));
+    _joints[Landmarks.Spine] = new Joint(_anim.GetBoneTransform(HumanBodyBones.Spine));
+
+    // Connections
+    // Right Arm
+    _joints[Landmarks.RightShoulder].Child = _joints[Landmarks.RightElbow];
+    _joints[Landmarks.RightElbow].Child = _joints[Landmarks.RightWrist];
+    _joints[Landmarks.RightElbow].Parent = _joints[Landmarks.RightShoulder];
+
+    // Left Arm
+    _joints[Landmarks.LeftShoulder].Child = _joints[Landmarks.LeftElbow];
+    _joints[Landmarks.LeftElbow].Child = _joints[Landmarks.LeftWrist];
+    _joints[Landmarks.LeftElbow].Parent = _joints[Landmarks.LeftShoulder];
+
+    // Right Leg
+    _joints[Landmarks.RightHip].Child = _joints[Landmarks.RightKnee];
+    _joints[Landmarks.RightKnee].Child = _joints[Landmarks.RightAnkle];
+    _joints[Landmarks.RightAnkle].Child = _joints[Landmarks.RightFootIndex];
+    _joints[Landmarks.RightAnkle].Parent = _joints[Landmarks.RightKnee];
+
+    // Left Leg
+    _joints[Landmarks.LeftHip].Child = _joints[Landmarks.LeftKnee];
+    _joints[Landmarks.LeftKnee].Child = _joints[Landmarks.LeftAnkle];
+    _joints[Landmarks.LeftAnkle].Child = _joints[Landmarks.LeftFootIndex];
+    _joints[Landmarks.LeftAnkle].Parent = _joints[Landmarks.LeftKnee];
+
+    // Spine
+    _joints[Landmarks.Spine].Child = _joints[Landmarks.Neck];
+    _joints[Landmarks.Neck].Child = _joints[Landmarks.Head];
+
+    // Set Inverse
+    Vector3 forward = GetNormal(_joints[Landmarks.LeftShoulder].position, _joints[Landmarks.LeftHip].position, _joints[Landmarks.RightHip].position);
+    foreach (Joint joint in _joints) {
+      if (joint != null && joint.Child != null) {
+        joint.Forward = Quaternion.LookRotation(joint.position - joint.Child.position, forward);
+      }
+    }
+    Joint hip = _joints[Landmarks.Hip];
+    hip.Forward = Quaternion.LookRotation(forward);
+
+    // For Head Rotation
+    Joint head = _joints[Landmarks.Head];
+    Vector3 gaze = _joints[Landmarks.Nose].position - _joints[Landmarks.Head].position;
+    head.Forward = Quaternion.LookRotation(gaze);  
   }
 
   public void Update() {
-    // Test code
-    _hip.SetRotation(new Vector3(-1, 0, 0), Vector3.up);
+  }
+
+  public void SetPoseLandmarks(NormalizedLandmarkList poseLandmarks) {
+  }
+
+  // <summary>
+  // Get the normal to a triangle from the three corner points, a, b and c.
+  // See https://docs.unity3d.com/ScriptReference/Vector3.Cross.html.
+  // </summary>
+  private Vector3 GetNormal(Vector3 a, Vector3 b, Vector3 c) {
```

Comment:
Just to make sure you have already considered the corner case: do we need to ensure that the passed-in arguments can actually form a triangle? Otherwise, the result of GetNormal() could be zero if the vector b-a and c-a are in the same direction.

If GetNormal() is a general-purpose method, it'd be ideal to check-and-handle or assert on this corner case.

And you can also consider extracting GetNormal() to a static Utils or MathUtils class since it looks not specific to ModelAnimator.

Reply:
Good point. Added an assertion to make the error more prominent.

I would hold off the suggestion to extract to utility for now, since doing this I wish:
1. We have more general utility aspect of the class, rather than adding a single-method util that only used by a single instance.
2. Add proper test coverage. I haven't get the test framework setup yet.

Thanks for the suggestion.